clustered/common/src/main/java/org/ehcache/clustered/common/messages/ServerStoreOpCodec.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.common.messages;
+
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+
+import static org.ehcache.clustered.common.messages.ServerStoreOpMessage.AppendMessage;
+import static org.ehcache.clustered.common.messages.ServerStoreOpMessage.GetAndAppendMessage;
+import static org.ehcache.clustered.common.messages.ServerStoreOpMessage.ReplaceAtHeadMessage;
+
+/**
+ *
+ */
+public class ServerStoreOpCodec {
+
+  private static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  private static final byte MSG_TYPE_OFFSET = 1;
+  private static final byte STORE_OP_CODE_OFFSET = 1;
+  private static final byte CACHE_ID_LEN_OFFSET = 4;
+  private static final byte KEY_OFFSET = 8;
+  private static final byte CHAIN_LEN_OFFSET = 4;
+
+  public static byte[] encode(ServerStoreOpMessage message) {
```

Comment:
Had a discussion with @albinsuresh @rkavanap @itsroshan & sairam and no one seems to like this switch/case thing. This needs refactoring.